tests/apollo/test_skvbc_restart_recovery.py

```diff
@@ -348,3 +348,42 @@ async def test_restarting_f_replicas_for_view_change(self, bft_network):
         await bft_network.wait_for_fast_path_to_be_prevalent(
             run_ops=lambda: skvbc.run_concurrent_ops(num_ops=20, write_weight=1), threshold=20)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: c == 0, rotate_keys=True)
+    @verify_linearizability()
+    async def test_recovering_of_primary_with_initiated_view_change(self, bft_network, tracker):
+        """
+        The Apollo test, which should be part of the test_skvbc_restart_recovery suite needs to implement the following steps:
+
+        1) Start all Replicas.
+        2) Stop the expected next Primary.
+        3) Advance the system 5 Checkpoints.
+        4) Stop the current Primary.
+        5) Send Client operations in order to trigger View Change.
+        6) Bring back up all the previously stopped replicas.
+        7) Verify View Change was successful.
+        8) Goto Step 2.
+        """
+
+        # start replicas
+        [bft_network.start_replica(i) for i in bft_network.all_replicas()]
+
+        # Stop the expected next Primary.
+        bft_network.stop_replica(1)
+
+        # Advance the system 5 Checkpoints.
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        await skvbc.run_concurrent_ops(150 * 5)
+
+        # Stop the current Primary
+        bft_network.stop_replica(0)
+
+        # Send Client operations in order to trigger View Change.
+        await skvbc.run_concurrent_ops(10)
+
+        #Bring back up all the previously stopped replicas.
+        bft_network.start_replica(0)
+        bft_network.start_replica(1)
+
+        #Verify View Change was successful.
+        await bft_network.wait_for_replicas_to_reach_view(bft_network.all_replicas(), 2)
```

Comment:
As per step 8) defined above - isn't this sequence supposed to run in a loop?

Reply:
Just as a note to Teo's comment, please bear in mind that if this sequence runs in a loop the next expected primary will no longer be 1.
If this is the case, calling `await bft_network.get_current_primary()` might be helpful.

Reply:
how many times it should run?